### PR TITLE
Add support for speculative query execution.

### PR DESCRIPTION
**Pull request by nicholaspeshek**
Adding support for controlling client-side constant-sized speculative executions

### CODE REVIEW

File: inc/priam/cluster.hpp [hunk 1]
```diff
@@ -107,6 +107,18 @@ class cluster
...
 
+
+    /**
+     * Enables constant speculative executions for requests sent to the Cluster.
+     * @param delay Controls the length of time before sending a speculative request
+     * @param max_executions The maximum number of speculative requests to send.
+                             This should not be higher than your max replication factor.
+     * @return True if the heartbeat intervals were updated.
+     */
+    auto cluster::speculative_execution(
```
Comment (jbaldwin) [inline inc/priam/cluster.hpp]: shouldn't need the namespace in the header declaration

File: inc/priam/cluster.hpp [hunk 2]
```diff
@@ -107,6 +107,18 @@ class cluster
         std::chrono::milliseconds update_rate,
         uint64_t                  min_measured) -> bool;
 
+
+    /**
+     * Enables constant speculative executions for requests sent to the Cluster.
+     * @param delay Controls the length of time before sending a speculative request
+     * @param max_executions The maximum number of speculative requests to send.
+                             This should not be higher than your max replication factor.
+     * @return True if the heartbeat intervals were updated.
```
Comment (jbaldwin) [inline inc/priam/cluster.hpp]: this  @return comment might need an update

File: inc/priam/cluster.hpp [hunk 3]
```diff
@@ -107,6 +107,18 @@ class cluster
...
+     */
+    auto cluster::speculative_execution(
+        std::chrono::milliseconds delay,
+        int                       max_executions) -> bool;
```
Comment (jbaldwin) [inline inc/priam/cluster.hpp]: Can we make this type maybe like uint16_t?  The docs don't say negative numbers are valid, you can just static_cast<int>(max_executions) when passing into the cassandra library

File: src/cluster.cpp [hunk 4]
```diff
@@ -93,6 +93,22 @@ auto cluster::latency_aware_routing(
...
 
+auto cluster::speculative_execution(
+    std::chrono::milliseconds delay,
+    int                       max_executions) -> bool
+{
+    if (m_cass_cluster_ptr != nullptr)
+    {
+        CassError error = cass_cluster_set_constant_speculative_execution_policy(
+            m_cass_cluster_ptr.get(),
+            static_cast<cass_uint64_t>(delay.count()),
```
Comment (jbaldwin) [inline src/cluster.cpp:104]: std::chrono::milliseconds can be negative I think, might want to check and return false if invalid

**Review by jbaldwin — CHANGES_REQUESTED** [review]

File: src/cluster.cpp (hunk 4 above)
Comment (nicholaspeshek) [reply src/cluster.cpp:104]: Added a greater than zero check

File: inc/priam/cluster.hpp (hunk 2 above)
Comment (nicholaspeshek) [reply inc/priam/cluster.hpp]: Definitely does. Corrected.

File: inc/priam/cluster.hpp (hunk 1 above)
Comment (nicholaspeshek) [reply inc/priam/cluster.hpp]: That was a copy/paste fail. Fixed.

File: inc/priam/cluster.hpp (hunk 3 above)
Comment (nicholaspeshek) [reply inc/priam/cluster.hpp]: Seems reasonable. Think I got that one.

**Review by jbaldwin — APPROVED** [review]

**Pull request merged**